src/DepotDelivery.jl

```diff
@@ -27,29 +27,53 @@ function sandbox(f::Function)
     end
 end
 
-#-----------------------------------------------------------------------------# build
-function build(path::String; platform = Base.BinaryPlatforms.HostPlatform(), verbose=true)
+#-----------------------------------------------------------------------------#
+"""
+Builds the depot for a specified project.
+
+Arguments:
+- `path::String`: The path to the project directory containing `Project.toml` or `JuliaProject.toml`.
+- `platform::AbstractPlatform`: The target platform for building (default is the host platform).
+- `verbose::Bool`: Whether to display verbose output during the build process (default is `true`).
+- `depot::String`: The path to the depot directory (default is a temporary directory).
+- `precompiled::Bool`: Whether to enable precompilation of packages (default is `false`).
+
+Returns:
+- The path to the built depot.
+
+Example:
+```julia
+depot_path = build("/path/to/your/project")
+"""
+function build(path::String; platform = Base.BinaryPlatforms.HostPlatform(), verbose=true, depot = mktempdir(), precompiled=false)
     path = abspath(path)
-    depot = mktempdir()
+    mkpath(depot)
     sandbox() do
         proj_file = joinpath(path, "Project.toml")
         proj_file = isfile(proj_file) ? proj_file : joinpath(path, "JuliaProject.toml")
         isfile(proj_file) || error("No Project.toml or JuliaProject.toml found in `$path`.")
         proj = TOML.parsefile(proj_file)
-        name = proj["name"]
+        # Defines a project name for Project.toml that don't come from a package
+        name = haskey(proj, "name") ? proj["name"] : Base.basename(Base.dirname(proj_file))
         build_spec = Dict(
             :datetime => Dates.now(),
             :versioninfo => sprint(InteractiveUtils.versioninfo),
             :project_file => proj_file,
             :project => proj,
             :platform => string(platform)
         )
-        mkdir(joinpath(depot, "config"))
-        mkdir(joinpath(depot, "dev"))
+        mkpath(joinpath(depot, "config"))
+        mkpath(joinpath(depot, "dev", name))
         push!(empty!(DEPOT_PATH), depot)
-        ENV["JULIA_PKG_PRECOMPILE_AUTO"] = "0"  # Needed when building for non-host platforms
+        
+        # Disabling precompile for non-host platforms
+        precompiled ? delete!(ENV, "JULIA_PKG_PRECOMPILE_AUTO") : ENV["JULIA_PKG_PRECOMPILE_AUTO"] = "0" 
 
-        cp(path, joinpath(depot, "dev", name))  # Copy project into dev/
+        for file in filter(f -> endswith(f, ".toml"), readdir(path))
+            # Copy only .toml files into dev/
+            cp(joinpath(path, file), joinpath(depot, "dev", name, file), force=true) 
+        end
+ 
```

Comment:
Why not `cp` the entire directory?  Julia expects full packages to be inside `dev/`

Reply:
You are right, for the use case of packages, it is necessary to copy everything. I rolled back to the original implementation.